### PR TITLE
SOC-895 | Introduce AuthView + canonicalUrl for Auth pages

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-895

### CODE REVIEW

File: server/facets/auth/join.ts [hunk 1]
```diff
@@ -1,50 +1,38 @@
 /// <reference path='../../../typings/hapi/hapi.d.ts' />
 import authUtils = require('../../lib/AuthUtils');
 import caching = require('../../lib/Caching');
+import authView = require('./authView');
+import deepExtend = require('deep-extend');
```
Comment (rogatty) [inline server/facets/auth/join.ts]:
```
server/facets/auth/join.ts(5,29): 2307 Cannot find external module 'deep-extend'.
Build contains some typescript errors/warnings
Build step 'Execute shell' marked build as failure
```

File: server/facets/auth/authView.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		return request.query.redirect || '/';
+	}
+
+	export function getCanonicalUrl (request: Hapi.Request): string {
+		return 'https://' + request.headers.host + request.path;
+	}
+
+	export function getDefaultContext (request: Hapi.Request): AuthViewContext {
+		return {
+			title: undefined,
```
Comment (bkoval) [inline server/facets/auth/authView.ts]: If it's undefined, there's no need to add the key here :)

**Comment by bkoval** [comment]
LGTM!

**Comment by jsutterfield** [comment]
LGTM

**Pull request merged**